src/main/java/com/amazonaws/kinesisvideo/producer/StreamInfo.java

```diff
@@ -237,14 +238,14 @@ public StreamInfo(final int version, @Nullable final String name, @Nonnull final
                       final long connectionStalenessDuration, final long timecodeScale, final boolean recalculateMetrics,
                       @Nullable final byte[] codecPrivateData,
                       @Nullable final Tag[] tags,
-                      @Nonnull final NalAdaptationFlags nalAdaptationFlags) {
+                      @Nonnull final NalAdaptationFlags nalAdaptationFlags, final boolean allowStreamCreation) {
         this(version, name, streamingType, contentType, kmsKeyId, retentionPeriod, adaptive, maxLatency,
                 fragmentDuration, keyFrameFragmentation, frameTimecodes, absoluteFragmentTimes, fragmentAcks,
                 recoverOnError, avgBandwidthBps, frameRate, bufferDuration, replayDuration,
                 connectionStalenessDuration, timecodeScale, recalculateMetrics, tags,
                 nalAdaptationFlags,
                 null,
-                new TrackInfo[] {new TrackInfo(DEFAULT_TRACK_ID, codecId, trackName, codecPrivateData, VIDEO)});
```

Comment:
For backwards compat. we need a new constructor.

Reply:
(this test will break: https://github.com/awslabs/amazon-kinesis-video-streams-producer-sdk-java/blob/f45428b3d730ae30fe63ca025cc60013357a520d/src/test/java/com/amazonaws/kinesisvideo/common/ProducerTestBase.java#L205)